Core/src/main/java/com/plotsquared/core/generator/HybridUtils.java

```diff
@@ -594,92 +643,100 @@ public boolean regenerateRoad(final PlotArea area, final BlockVector2 chunk, int
+                                if (block != null) {
                                     queue.setBlock(
                                             finalX + X + plotWorld.ROAD_OFFSET_X,
                                             minY + dy,
                                             finalZ + Z + plotWorld.ROAD_OFFSET_Z,
-                                            WEExtent.AIRBASE
+                                            block
                                     );
                                 } else {
-                                    BaseBlock block = blocks[dy];
-                                    if (block != null) {
-                                        queue.setBlock(
-                                                finalX + X + plotWorld.ROAD_OFFSET_X,
-                                                minY + dy,
-                                                finalZ + Z + plotWorld.ROAD_OFFSET_Z,
-                                                block
-                                        );
-                                    } else {
-                                        queue.setBlock(
-                                                finalX + X + plotWorld.ROAD_OFFSET_X,
-                                                minY + dy,
-                                                finalZ + Z + plotWorld.ROAD_OFFSET_Z,
-                                                WEExtent.AIRBASE
-                                        );
-                                    }
+                                    queue.setBlock(
+                                            finalX + X + plotWorld.ROAD_OFFSET_X,
+                                            minY + dy,
+                                            finalZ + Z + plotWorld.ROAD_OFFSET_Z,
+                                            WEExtent.AIRBASE
+                                    );
                                 }
                             }
-                            BiomeType biome = plotWorld.G_SCH_B.get(MathMan.pair(absX, absZ));
-                            if (biome != null) {
-                                queue.setBiome(finalX + X + plotWorld.ROAD_OFFSET_X, finalZ + Z + plotWorld.ROAD_OFFSET_Z, biome);
-                            } else {
-                                queue.setBiome(
-                                        finalX + X + plotWorld.ROAD_OFFSET_X,
-                                        finalZ + Z + plotWorld.ROAD_OFFSET_Z,
-                                        plotWorld.getPlotBiome()
-                                );
-                            }
+                        }
+                        BiomeType biome = plotWorld.G_SCH_B.get(MathMan.pair(absX, absZ));
+                        if (biome != null) {
+                            queue.setBiome(finalX + X + plotWorld.ROAD_OFFSET_X, finalZ + Z + plotWorld.ROAD_OFFSET_Z, biome);
+                        } else {
+                            queue.setBiome(
```

Comment:
This uses the now deprecated `#setBiome`, looking on the amount of args. Do we not support 3D biomes here?

Reply:
Somewhat out of the scope of the PR; 3D biomes aren't supported anywhere else in the plugin

Reply:
Oic, good to know 👍🏻